hazelcast/src/test/java/com/hazelcast/instance/NodeEngineThreadLeakTest.java

```diff
@@ -38,18 +24,31 @@
 import com.hazelcast.spi.impl.eventservice.impl.EventServiceImpl;
 import com.hazelcast.spi.impl.executionservice.impl.ExecutionServiceImpl;
 import com.hazelcast.spi.impl.operationparker.impl.OperationParkerImpl;
+import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Set;
+
+import static classloading.ThreadLeakTestUtils.assertHazelcastThreadShutdown;
+import static classloading.ThreadLeakTestUtils.getThreads;
+import static org.junit.Assert.fail;
 
 /**
- * Tests that exception in Node constructor leads to properly finished services.
+ * Tests that an exception in the {@link NodeEngineImpl} constructor leads to properly finished services.
  */
 @RunWith(PowerMockRunner.class)
-@PrepareForTest({ DefaultNodeContext.class, NodeEngineImpl.class })
+@PrepareForTest({DefaultNodeContext.class, NodeEngineImpl.class})
 @Category(QuickTest.class)
-public class NodeThreadLeakTest {
+public class NodeEngineThreadLeakTest extends HazelcastTestSupport {
```

Comment:
Renaming is not necessary here. The ur-issue was not in the `NodeEngine`, but in call `nodeContext.createConnectionManager(this, serverSocketChannel)` within the `Node.<init>`. It fails when the SSL/TLS is not configured properly.